### DISCUSSION TITLE
Convert PDF > GIF creates artefacts on line graphs

### DISCUSSION THREAD

**Discussion by jkorff** [ImageMagick/ImageMagick#1920] [General] [answered]
I'm using ImageMagick 7.0.7-25 portable x86 (due to restricted IT environment) and need to convert high-res graph PDFs to large GIFs (2000 px wide).

I'm using a PowerShell script that calls Magick's convert.exe with the following parameters:

`convert.exe -colorspace srgb -density 240 source.pdf -flatten destination.gif
`

Several graphs, especially line graphs, show artefacts that look like scratching (check out especially the bottom spikes of the blue line):
![gif](https://user-images.githubusercontent.com/970056/80038165-5fcaa400-8538-11ea-90ea-62ef010609c0.PNG)

But the original looks ok:
![pdf](https://user-images.githubusercontent.com/970056/80038461-0adb5d80-8539-11ea-8680-2c3b3360b4a6.PNG)
(these are screenshots from the images)

Is there a way to improve the output? I've played with the `density` parameter but wasn't able to remove the artefacts I'm seeing.

Thank you!

**Comment by urban-warrior** [comment]
Try this command:

`convert.exe -colorspace srgb -density 480 source.pdf -resize 50% -flatten destination.gif`

Change to -density 960, -resize 25% for additional improvement.

**Comment by fmw42** [comment]
Post one of your PDFs if the suggestion by urban-warrior does not help. Also provide your Ghostscript version. Some versions are known to be buggy.

**Comment by jkorff** [comment]
Thank you @urban-warrior, those modified settings improved the quality. The 480/50% setting still leaves a little disturbance behind, but is fast; the 960/25% setting improves further, but is much slower. I'll have to make a call which setting we can use feasibly in production.

@fmw42, the Ghostscript version is 9.22.

**Comment by snibgo** [comment, marked as answer]
I think GS 9.22 is a one of the versions with known problems. I suggest you try with an older version (eg 9.19) or the current version.

If you paste a link to a sample PDF, we can try with different GS versions.

**Comment by jkorff** [comment]
Thank you, @snibgo and @fmw42, you were spot on. After downgrading to Ghostscript 9.19 portable and re-running the task with my original parameters I got a much better result, even better than with the 960 density setting:

![gif-n](https://user-images.githubusercontent.com/970056/80055105-3a538f80-8564-11ea-9d98-e66285bcc355.PNG)

I can work with this result. Thank you very much everyone.
